fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -202,7 +202,8 @@ void FixedLagSmoother::optimizationLoop()
       // Update the graph
       graph_->update(*new_transaction);
       // Optimize the entire graph
-      graph_->optimize(params_.solver_options);
+      summary_ = graph_->optimize(params_.solver_options);
+
```

Comment:
BTW, not from this PR, but in https://github.com/locusrobotics/fuse/blob/df09ed00b456fca8834e91138df401fc343fff86/fuse_optimizers/src/fixed_lag_smoother.cpp#L194 we skip the optimization cycle if there are no pending transactions. However, this means that we also don't update and optimize the graph with the marginal transaction prepared in the previous cycle.

That's probably fine, but if for some reasons we don't get transactions from the sensor models for a few cycles, I wonder if that could cause any issues. Any thoughts?

Reply:
If the incoming transaction is empty, then there is nothing to advance the fixed-lag window forward in time. We _could_ apply just the marginal transaction and reoptimize, but we would end up with the exact same solution as the previous cycle. So I think it's fine the way it is.

Now, it might make other things, like tf, unhappy if there is no new data for several seconds. But that's a different problem.

Reply:
That's what I thought. I agree it's fine as is. :+1: